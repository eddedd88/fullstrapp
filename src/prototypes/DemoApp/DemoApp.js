import React, { Component, Fragment } from 'react'
import BottomNavBar from '../../components/BottomNavBar'
import { Switch, Route, Redirect } from 'react-router'
import FeedPageContainer from '../../containers/FeedPageContainer'
import GridPageContainer from '../../containers/GridPageContainer'
import ProfilePageContainer from '../../containers/ProfilePageContainer'
import paths from '../../routes/paths'
import Onboarding from '../Onboarding'
import FeedItemPageContainer from '../../containers/FeedItemPageContainer'
import GridIcon from '@material-ui/icons/GridOn'
import PersonIcon from '@material-ui/icons/Person'
import LocationCityIcon from '@material-ui/icons/LocationCity'

const onboardingKey = 'alreadyOnboarded'
type State = {
  needOnboarding: boolean
}

class App extends Component<{||}, State> {
  state = {
    needOnboarding: !window.localStorage.getItem(onboardingKey)
  }

  handleFinishOnboarding = () => {
    window.localStorage.setItem(onboardingKey, true)
    this.setState({
      needOnboarding: false
    })
  }

  render () {
    const { needOnboarding } = this.state

    return needOnboarding ? (
      <Onboarding onDone={this.handleFinishOnboarding} />
    ) : (
      <Fragment>
        <div style={{ marginBottom: 64 }}>
          <Switch>
            <Route path={paths.feedItem} component={FeedItemPageContainer} />
            <Route path={paths.feed} component={FeedPageContainer} />
            <Route path={paths.grid} component={GridPageContainer} />
            <Route
              path={paths.profileFeedItem}
              component={FeedItemPageContainer}
            />
            <Route path={paths.profile} component={ProfilePageContainer} />
            <Redirect to={paths.feed} />
          </Switch>
        </div>
        <BottomNavBar
          items={[
            {
              label: 'Feed',
              icon: <LocationCityIcon />,
              path: paths.feed
            },
            {
              label: 'Grid',
              icon: <GridIcon />,
              path: paths.grid
            },
            {
              label: 'Profile',
              icon: <PersonIcon />,
              path: paths.profile
            }
          ]}
        />
      </Fragment>
    )
  }
}

export default App
