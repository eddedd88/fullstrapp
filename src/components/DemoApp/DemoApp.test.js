import React from 'react'
import ReactDOM from 'react-dom'
import DemoApp from './DemoApp'
import { BrowserRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'

describe('components > DemoApp', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<DemoApp />)
    expect(tree.toJSON()).toMatchSnapshot()
  })
})
