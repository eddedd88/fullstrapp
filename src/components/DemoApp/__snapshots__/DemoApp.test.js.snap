// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components > DemoApp renders correctly 1`] = `
Array [
  <header
    className="MuiPaper-root-12 MuiPaper-elevation4-18 MuiAppBar-root-4 MuiAppBar-positionStatic-8 MuiAppBar-colorPrimary-10"
>
    <div
        className="MuiToolbar-root-39 MuiToolbar-regular-41 MuiToolbar-gutters-40"
    >
        <h2
            className="MuiTypography-root-43 MuiTypography-title-49 MuiTypography-colorInherit-62"
            style={
                Object {
                    "flex": "auto",
                  }
            }
        >
            Getting Started
        </h2>
        <button
            className="MuiButtonBase-root-73 MuiIconButton-root-67 MuiIconButton-colorInherit-68"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex="0"
            type="button"
        >
            <span
                className="MuiIconButton-label-72"
            >
                <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-76"
                    color={undefined}
                    focusable="false"
                    viewBox="0 0 24 24"
                >
                    <path
                        d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                    />
                    <path
                        d="M0 0h24v24H0z"
                        fill="none"
                    />
                </svg>
            </span>
            <span
                className="MuiTouchRipple-root-83"
            />
        </button>
        <div
            className="MuiPaper-root-12 MuiPaper-elevation1-15"
            color="inherit"
            style={
                Object {
                    "left": 0,
                    "opacity": 0,
                    "overflow": "hidden",
                    "pointerEvents": "none",
                    "position": "absolute",
                    "transition": "opacity 100ms linear",
                    "width": "100%",
                  }
            }
        >
            <div
                className="MuiToolbar-root-39 MuiToolbar-regular-41"
            >
                <button
                    className="MuiButtonBase-root-73 MuiIconButton-root-67"
                    disabled={false}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex="0"
                    type="button"
                >
                    <span
                        className="MuiIconButton-label-72"
                    >
                        <svg
                            aria-hidden="true"
                            className="MuiSvgIcon-root-76"
                            color={undefined}
                            focusable="false"
                            viewBox="0 0 24 24"
                        >
                            <path
                                d="M0 0h24v24H0z"
                                fill="none"
                            />
                            <path
                                d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
                            />
                        </svg>
                    </span>
                    <span
                        className="MuiTouchRipple-root-83"
                    />
                </button>
                <div
                    className="MuiInput-root-90 MuiInput-fullWidth-97"
                >
                    <input
                        aria-invalid={undefined}
                        autoComplete={undefined}
                        autoFocus={undefined}
                        className="MuiInput-input-98 MuiInput-inputType-101 MuiInput-inputTypeSearch-102"
                        defaultValue={undefined}
                        disabled={undefined}
                        id={undefined}
                        name={undefined}
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        onKeyDown={undefined}
                        onKeyUp={undefined}
                        placeholder="Search"
                        readOnly={undefined}
                        required={undefined}
                        rows={undefined}
                        type="search"
                        value=""
                    />
                    <div
                        className="MuiInputAdornment-root-103 MuiInputAdornment-positionEnd-105"
                    >
                        <button
                            aria-label="Clear search query"
                            className="MuiButtonBase-root-73 MuiIconButton-root-67"
                            disabled={false}
                            onBlur={[Function]}
                            onClick={[Function]}
                            onFocus={[Function]}
                            onKeyDown={[Function]}
                            onKeyUp={[Function]}
                            onMouseDown={[Function]}
                            onMouseLeave={[Function]}
                            onMouseUp={[Function]}
                            onTouchEnd={[Function]}
                            onTouchMove={[Function]}
                            onTouchStart={[Function]}
                            tabIndex="0"
                            type="button"
                        >
                            <span
                                className="MuiIconButton-label-72"
                            >
                                <svg
                                    aria-hidden="true"
                                    className="MuiSvgIcon-root-76"
                                    color={undefined}
                                    focusable="false"
                                    viewBox="0 0 24 24"
                                >
                                    <path
                                        d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                                    />
                                    <path
                                        d="M0 0h24v24H0z"
                                        fill="none"
                                    />
                                </svg>
                            </span>
                            <span
                                className="MuiTouchRipple-root-83"
                            />
                        </button>
                    </div>
                </div>
            </div>
        </div>
    </div>
</header>,
  <div
    className="Wrapper-root-107 WithStyles-Wrapper--root-106"
>
    <div
        className="MuiPaper-root-12 MuiPaper-elevation0-14 MuiStepper-root-108 MuiStepper-vertical-110"
    >
        <div
            className="MuiStep-root-112 MuiStep-vertical-114"
        >
            <span
                className="MuiStepLabel-root-116 MuiStepLabel-vertical-118"
            >
                <span
                    className="MuiStepLabel-iconContainer-124"
                >
                    <svg
                        aria-hidden="true"
                        className="MuiSvgIcon-root-76 MuiStepIcon-root-127 MuiStepIcon-active-129"
                        color={undefined}
                        focusable="false"
                        viewBox="0 0 24 24"
                    >
                        <circle
                            cx="12"
                            cy="12"
                            r="12"
                        />
                        <text
                            className="MuiStepIcon-text-128"
                            textAnchor="middle"
                            x="12"
                            y="16"
                        >
                            1
                        </text>
                    </svg>
                </span>
                <span
                    className="MuiStepLabel-labelContainer-126"
                >
                    <span
                        className="MuiTypography-root-43 MuiTypography-body1-52 MuiStepLabel-label-119 MuiStepLabel-active-120"
                    >
                        Select campaign settings
                    </span>
                </span>
            </span>
            <div
                className="MuiStepContent-root-132"
                disabled={false}
                icon={1}
            >
                <div
                    className="MuiCollapse-container-135 MuiCollapse-entered-136 MuiStepContent-transition-134"
                    style={
                        Object {
                            "minHeight": "0px",
                          }
                    }
                >
                    <div
                        className="MuiCollapse-wrapper-137"
                    >
                        <div
                            className="MuiCollapse-wrapperInner-138"
                        >
                            <p
                                className="MuiTypography-root-43 MuiTypography-body1-52"
                            >
                                For each ad campaign that you create, you can control how much
                                              you're willing to spend on clicks and conversions, which networks
                                              and geographical locations you want your ads to show on, and more.
                            </p>
                            <div
                                className="WebOnboarding-actionsContainer-2"
                            >
                                <div>
                                    <button
                                        className="MuiButtonBase-root-73 MuiButtonBase-disabled-74 MuiButton-root-139 MuiButton-text-141 MuiButton-flat-144 MuiButton-disabled-157 WebOnboarding-button-1"
                                        disabled={true}
                                        onBlur={[Function]}
                                        onClick={[Function]}
                                        onFocus={[Function]}
                                        onKeyDown={[Function]}
                                        onKeyUp={[Function]}
                                        onMouseDown={[Function]}
                                        onMouseLeave={[Function]}
                                        onMouseUp={[Function]}
                                        onTouchEnd={[Function]}
                                        onTouchMove={[Function]}
                                        onTouchStart={[Function]}
                                        tabIndex="-1"
                                        type="button"
                                    >
                                        <span
                                            className="MuiButton-label-140"
                                        >
                                            Back
                                        </span>
                                    </button>
                                    <button
                                        className="MuiButtonBase-root-73 MuiButton-root-139 MuiButton-contained-148 MuiButton-containedPrimary-149 MuiButton-raised-151 MuiButton-raisedPrimary-152 WebOnboarding-button-1"
                                        disabled={false}
                                        onBlur={[Function]}
                                        onClick={[Function]}
                                        onFocus={[Function]}
                                        onKeyDown={[Function]}
                                        onKeyUp={[Function]}
                                        onMouseDown={[Function]}
                                        onMouseLeave={[Function]}
                                        onMouseUp={[Function]}
                                        onTouchEnd={[Function]}
                                        onTouchMove={[Function]}
                                        onTouchStart={[Function]}
                                        tabIndex="0"
                                        type="button"
                                    >
                                        <span
                                            className="MuiButton-label-140"
                                        >
                                            Next
                                        </span>
                                        <span
                                            className="MuiTouchRipple-root-83"
                                        />
                                    </button>
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        </div>
        <div
            className="MuiStepConnector-root-163 MuiStepConnector-vertical-165"
        >
            <span
                className="MuiStepConnector-line-167 MuiStepConnector-lineVertical-169"
            />
        </div>
        <div
            className="MuiStep-root-112 MuiStep-vertical-114"
        >
            <span
                className="MuiStepLabel-root-116 MuiStepLabel-vertical-118 MuiStepLabel-disabled-123"
            >
                <span
                    className="MuiStepLabel-iconContainer-124"
                >
                    <svg
                        aria-hidden="true"
                        className="MuiSvgIcon-root-76 MuiStepIcon-root-127"
                        color={undefined}
                        focusable="false"
                        viewBox="0 0 24 24"
                    >
                        <circle
                            cx="12"
                            cy="12"
                            r="12"
                        />
                        <text
                            className="MuiStepIcon-text-128"
                            textAnchor="middle"
                            x="12"
                            y="16"
                        >
                            2
                        </text>
                    </svg>
                </span>
                <span
                    className="MuiStepLabel-labelContainer-126"
                >
                    <span
                        className="MuiTypography-root-43 MuiTypography-body1-52 MuiStepLabel-label-119"
                    >
                        Create an ad group
                    </span>
                </span>
            </span>
            <div
                className="MuiStepContent-root-132"
                disabled={true}
                icon={2}
            />
        </div>
        <div
            className="MuiStepConnector-root-163 MuiStepConnector-vertical-165"
        >
            <span
                className="MuiStepConnector-line-167 MuiStepConnector-lineVertical-169"
            />
        </div>
        <div
            className="MuiStep-root-112 MuiStep-vertical-114"
        >
            <span
                className="MuiStepLabel-root-116 MuiStepLabel-vertical-118 MuiStepLabel-disabled-123"
            >
                <span
                    className="MuiStepLabel-iconContainer-124"
                >
                    <svg
                        aria-hidden="true"
                        className="MuiSvgIcon-root-76 MuiStepIcon-root-127"
                        color={undefined}
                        focusable="false"
                        viewBox="0 0 24 24"
                    >
                        <circle
                            cx="12"
                            cy="12"
                            r="12"
                        />
                        <text
                            className="MuiStepIcon-text-128"
                            textAnchor="middle"
                            x="12"
                            y="16"
                        >
                            3
                        </text>
                    </svg>
                </span>
                <span
                    className="MuiStepLabel-labelContainer-126"
                >
                    <span
                        className="MuiTypography-root-43 MuiTypography-body1-52 MuiStepLabel-label-119"
                    >
                        Create an ad
                    </span>
                </span>
            </span>
            <div
                className="MuiStepContent-root-132 MuiStepContent-last-133"
                disabled={true}
                icon={3}
            />
        </div>
    </div>
</div>,
]
`;
